### PR TITLE
Collections, Users, Reactions and Enrichment

### PR DESCRIPTION
**Pull request by mircea-cosbuc**

### CODE REVIEW

File: lib/stream/reactions.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        lookup_value = params[:activity_id]
+      elsif params[:user_id]
+        lookup_field = "user_id"
+        lookup_value = params[:user_id]
+      end
+      unless lookup_field.empty?
+        params.delete(lookup_field.to_sym)
+      end
+
+      uri = "/reaction/#{lookup_field}/#{lookup_value}/#{kind}/"
```
Comment (pterk) [inline lib/stream/reactions.rb]: Won't this give double slashes at the end if kind isn't set?

Comment (mircea-cosbuc) [reply lib/stream/reactions.rb]: It does, the API is handling it correctly though

File: lib/stream/client.rb [hunk 2]
```diff
@@ -16,12 +16,12 @@ class Client
 
     if RUBY_VERSION.to_f >= 2.1
```
Comment (tbarbugli) [inline lib/stream/client.rb]: @mircea-cosbuc you can remove this version checking

File: lib/stream/client.rb [hunk 3]
```diff
@@ -77,8 +78,7 @@ def initialize(api_key = '', api_secret = '', app_id = nil, opts = {})
     # @return [Stream::Feed]
     #
     def feed(feed_slug, user_id)
-      token = @signer.sign(feed_slug, user_id)
-      Stream::Feed.new(self, feed_slug, user_id, token)
+      Stream::Feed.new(self, feed_slug, user_id)
     end
 
     def create_user_session_token(user_id, extra_data = {})
```
Comment (tbarbugli) [inline lib/stream/client.rb]: we are deprecating `create_user_session_token` in favor of `create_user_token`

File: lib/stream/feed.rb [hunk 4]
```diff
@@ -38,41 +36,44 @@ def valid_user_id(user_id)
       !user_id.to_s[/^[\w-]+$/].nil?
     end
 
-    def get(params = {})
-      uri = "/feed/#{@feed_url}/"
+    def get(enrich: false, reactions: nil, **params)
```
Comment (tbarbugli) [inline lib/stream/feed.rb]: @mircea-cosbuc ruby's support for kw arguments was changed a couple times, are you sure this works on all versions we support?

File: lib/stream/signedrequest.rb [hunk 5]
```diff
@@ -1,43 +0,0 @@
-require 'http_signatures'
```
Comment (tbarbugli) [inline lib/stream/signedrequest.rb]: I think you forgot to remove this from the requirements manifest (Gemfile?)

**Pull request merged**